### PR TITLE
Format all code using black

### PR DESCRIPTION
**Pull request by lihuoran**
# Description

## Linked issue(s)/Pull request(s)

- [issue_number](issue_link)

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [ ] New feature
- [ ] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [ ] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [ ] Windows
  - [ ] Mac OS
  - [ ] Linux
- Python version:
  - [ ] 3.6
  - [ ] 3.7
- Key information snapshot(s):

## Needs Follow Up Actions

- [ ] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: maro/cli/k8s/executors/k8s_aks_executor.py [hunk 1]
```diff
@@ -617,46 +587,22 @@ def _create_k8s_container_config(
...
-            },
-            {
-                "name": "COMPONENT_INDEX",
-                "value": f"{component_index}"
-            },
-            {
-                "name": "PYTHONUNBUFFERED",
-                "value": "0"
-            }
+            {"name": "CLUSTER_ID", "value": f"{self.cluster_id}"},
```
Comment (lihuoran) [inline maro/cli/k8s/executors/k8s_aks_executor.py:598]: mark

File: examples/citi_bike/online_lp/citi_bike_ilp.py [hunk 2]
```diff
@@ -113,23 +109,26 @@ def _add_constraints(self, problem: LpProblem, demand: np.ndarray, supply: np.nd
                 ), f"Fulfillment_Limit_T{decision_point}_S{station}"
                 # For intermediate variables.
                 problem += (
-                    self._transfer_from[decision_point][station] == lpSum(
+                    self._transfer_from[decision_point][station]
```
Comment (dongguangbin) [inline examples/citi_bike/online_lp/citi_bike_ilp.py:112]: I think ``==`` right after the variable is easier to understand.

File: examples/citi_bike/online_lp/citi_bike_ilp.py [hunk 3]
```diff
@@ -113,23 +109,26 @@ def _add_constraints(self, problem: LpProblem, demand: np.ndarray, supply: np.nd
...
+                    == lpSum(
                         self._transfer[decision_point][station][neighbor_idx]
                         for neighbor_idx in range(self._num_neighbor)
                     )
                 ), f"TotalTransferFrom_T{decision_point}_S{station}"
                 problem += (
-                    self._transfer_to[decision_point][station] == lpSum(
+                    self._transfer_to[decision_point][station]
```
Comment (dongguangbin) [inline examples/citi_bike/online_lp/citi_bike_ilp.py:119]: I think ``==`` right after the variable is easier to understand.

File: examples/citi_bike/online_lp/citi_bike_ilp.py [hunk 4]
```diff
@@ -113,23 +109,26 @@ def _add_constraints(self, problem: LpProblem, demand: np.ndarray, supply: np.nd
...
-                        if station in self._station_neighbor_list[neighbor][:self._num_neighbor]
+                        if station in self._station_neighbor_list[neighbor][: self._num_neighbor]
                     )
                 ), f"TotalTransferTo_T{decision_point}_S{station}"
 
         for decision_point in range(1, self._num_decision_point):
             for station in range(self._num_station):
                 problem += (
-                    self._inventory[decision_point][station] == (
+                    self._inventory[decision_point][station]
```
Comment (dongguangbin) [inline examples/citi_bike/online_lp/citi_bike_ilp.py:130]: I think ``==`` right after the variable is easier to understand.

File: examples/citi_bike/online_lp/citi_bike_ilp.py [hunk 5]
```diff
@@ -138,7 +137,8 @@ def _add_constraints(self, problem: LpProblem, demand: np.ndarray, supply: np.nd
                     )
                 ), f"Inventory_T{decision_point}_S{station}"
                 problem += (
-                    self._safety_inventory[decision_point][station] <= (
+                    self._safety_inventory[decision_point][station]
```
Comment (dongguangbin) [inline examples/citi_bike/online_lp/citi_bike_ilp.py:140]: I think ``<=`` right after the variable is easier to understand.

File: examples/cim/rl/config.py [hunk 6]
```diff
@@ -1,11 +1,7 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
-env_conf = {
-    "scenario": "cim",
-    "topology": "toy.4p_ssdd_l0.0",
-    "durations": 560
-}
+env_conf = {"scenario": "cim", "topology": "toy.4p_ssdd_l0.0", "durations": 560}
```
Comment (Jinyu-W) [inline examples/cim/rl/config.py:4]: mark

File: examples/cim/rl/config.py [hunk 7]
```diff
@@ -15,29 +11,20 @@
...
-    "time_decay": 0.97
-}
+reward_shaping_conf = {"time_window": 99, "fulfillment_factor": 1.0, "shortage_factor": 1.0, "time_decay": 0.97}
 
 # obtain state dimension from a temporary env_wrapper instance
-state_dim = (
-    (state_shaping_conf["look_back"] + 1) * (state_shaping_conf["max_ports_downstream"] + 1) * len(port_attributes)
-    + len(vessel_attributes)
-)
+state_dim = (state_shaping_conf["look_back"] + 1) * (state_shaping_conf["max_ports_downstream"] + 1) * len(
```
Comment (Jinyu-W) [inline examples/cim/rl/config.py:25]: mark

File: examples/citi_bike/online_lp/launcher.py [hunk 8]
```diff
@@ -252,10 +256,10 @@ def get_action_list(self, env_tick: int, init_inventory: np.ndarray, finished_ev
         else:
             action = agent.get_action_list(
                 env_tick=env.tick,
-                init_inventory=env.snapshot_list["stations"][
-                    env.frame_index:env.agent_idx_list:"bikes"
-                ].astype(np.int16),
-                finished_events=env.get_finished_events()
+                init_inventory=env.snapshot_list["stations"][env.frame_index : env.agent_idx_list : "bikes"].astype(
```
Comment (Jinyu-W) [inline examples/citi_bike/online_lp/launcher.py:259]: mark

File: examples/cim/rl/env_sampler.py [hunk 9]
```diff
@@ -49,7 +58,9 @@ def _translate_to_env_action(
             actual_action = min(round(percent * action_scope.load), vsl_space)
         elif model_action > zero_action_idx:
             action_type = ActionType.DISCHARGE
-            early_discharge = vsl_snapshots[self._env.tick:vsl_idx:"early_discharge"][0] if has_early_discharge else 0
+            early_discharge = (
+                vsl_snapshots[self._env.tick : vsl_idx : "early_discharge"][0] if has_early_discharge else 0
+            )
```
Comment (dongguangbin) [inline examples/cim/rl/env_sampler.py:63]: mark

File: examples/vector_env/hello.py [hunk 10]
```diff
@@ -28,7 +30,7 @@ class VectorEnvUsage(Enum):
 
                     # Showcase: how to access information from snapshot list in vector env.
                     if env0_dec:
-                        ss0 = env.snapshot_list["vessels"][env0_dec.tick:env0_dec.vessel_idx:"remaining_space"]
+                        ss0 = env.snapshot_list["vessels"][env0_dec.tick : env0_dec.vessel_idx : "remaining_space"]
```
Comment (Jinyu-W) [inline examples/vector_env/hello.py:33]: Mark.

Compared with line 83 in examples/simulator/data_model.py

File: maro/cli/data_pipeline/vm_scheduling.py [hunk 11]
```diff
@@ -52,13 +52,7 @@ def __init__(self, topology: str, source: str, sample: int, seed: int, is_temp:
...
         self._clean_cpu_readings_file_name_list = []
 
-        self.aria2 = aria2p.API(
-            aria2p.Client(
-                host="http://localhost",
-                port=6800,
-                secret=""
-            )
-        )
+        self.aria2 = aria2p.API(aria2p.Client(host="http://localhost", port=6800, secret=""))
```
Comment (Jinyu-W) [inline maro/cli/data_pipeline/vm_scheduling.py:55]: Looks like it does not exceed the line length limitation. So what's the standard of breaking lines?

File: maro/cli/inspector/cim_dashboard.py [hunk 12]
```diff
@@ -59,26 +56,14 @@ def render_inter_view(source_path: str, epoch_num: int):
...
     data = helper.read_detail_csv(os.path.join(source_path, GlobalFileNames.ports_sum)).iloc[down_pooling_range]
-    data["remaining_space"] = list(
-        map(
-            lambda x, y, z: x - y - z,
-            data["capacity"],
-            data["full"],
-            data["empty"]
-        )
-    )
+    data["remaining_space"] = list(map(lambda x, y, z: x - y - z, data["capacity"], data["full"], data["empty"]))
```
Comment (dongguangbin) [inline maro/cli/inspector/cim_dashboard.py:63]: mark

File: maro/cli/inspector/cim_dashboard.py [hunk 13]
```diff
@@ -610,24 +547,12 @@ def _generate_top_k_summary(data: pd.DataFrame, snapshot_index: int, index_name_
...
+        map(lambda x, y: round(x / (y + 1 / 1000), 4), data_summary["acc_fulfillment"], data_summary["acc_booking"])
     )
 
-    data_summary["port name"] = list(
-        map(
-            lambda x: index_name_conversion.loc[int(x[6:])][0],
-            data_summary["name"]
-        )
-    )
+    data_summary["port name"] = list(map(lambda x: index_name_conversion.loc[int(x[6:])][0], data_summary["name"]))
```
Comment (dongguangbin) [inline maro/cli/inspector/cim_dashboard.py:553]: mark

File: maro/cli/inspector/citi_bike_dashboard.py [hunk 14]
```diff
@@ -250,31 +217,30 @@ def _generate_intra_view_by_snapshot(
...
+        data_filtered["Station Name"] = data_filtered["name"].apply(lambda x: index_name_conversion.loc[int(x)])
+    data_melt = data_filtered.melt(["Station Name", "name"], var_name="Attributes", value_name="Count")
+    intra_snapshot_line_chart = (
+        alt.Chart(data_melt)
+        .mark_bar()
+        .encode(
+            x=alt.X("name:N", axis=alt.Axis(title="Name")),
+            y="Count:Q",
+            color="Attributes:N",
+            tooltip=["Attributes", "Count", "Station Name"],
```
Comment (dongguangbin) [inline maro/cli/inspector/citi_bike_dashboard.py:229]: mark

File: maro/cli/grass/lib/services/node_agent/agent.py [hunk 15]
```diff
@@ -382,30 +368,29 @@ def load_images(self) -> None:
         for image_file_name, image_file_details in name_to_image_file_details_in_master.items():
             if (
                 image_file_name not in name_to_image_file_details_in_node
-                or name_to_image_file_details_in_node[image_file_name]["md5_checksum"] !=
-                image_file_details["md5_checksum"]
+                or name_to_image_file_details_in_node[image_file_name]["md5_checksum"]
```
Comment (Jinyu-W) [inline maro/cli/grass/lib/services/node_agent/agent.py:371]: anyway to add  `()` in this case?

File: maro/cli/grass/lib/services/node_agent/agent.py [hunk 16]
```diff
@@ -382,30 +368,29 @@ def load_images(self) -> None:
...
+            pool.starmap(self._load_image, params)
 
         with self._redis_controller.lock(f"lock:name_to_node_details:{self._local_node_details['name']}"):
             node_details = self._redis_controller.get_node_details(node_name=self._local_node_details["name"])
             # Update with mapping in master.
             node_details["image_files"] = name_to_image_file_details_in_master
             self._redis_controller.set_node_details(
-                node_name=self._local_node_details["name"],
-                node_details=node_details
+                node_name=self._local_node_details["name"], node_details=node_details
```
Comment (Jinyu-W) [inline maro/cli/grass/lib/services/node_agent/agent.py:393]: Add tailing `,` or not?

File: maro/cli/grass/lib/services/utils/docker_controller.py [hunk 17]
```diff
@@ -86,7 +84,7 @@ def create_container_with_config(create_config: dict) -> dict:
 
     @staticmethod
     def list_container_names() -> list:
-        command = "sudo docker ps -a --format \"{{.Names}}\""
+        command = 'sudo docker ps -a --format "{{.Names}}"'
```
Comment (Jinyu-W) [inline maro/cli/grass/lib/services/utils/docker_controller.py:87]: mark

File: maro/cli/inspector/dashboard_helper.py [hunk 18]
```diff
@@ -253,11 +237,7 @@ def _formula_define(data_original: pd.DataFrame) -> dict:
             data_right = _judge_append_data(data_original.head(0), res)
             if data_right:
                 data_original[f"{res[0]}+{res[1]}"] = list(
-                    map(
-                        lambda x, y: x + y,
-                        data_original[res[0]],
-                        data_original[res[1]]
-                    )
+                    map(lambda x, y: x + y, data_original[res[0]], data_original[res[1]])
```
Comment (dongguangbin) [inline maro/cli/inspector/dashboard_helper.py:240]: mark

File: maro/cli/inspector/env_data_process.py [hunk 19]
```diff
@@ -158,48 +159,45 @@ def _generate_summary(scenario: GlobalScenarios, source_path: str, prefix: str,
...
             input_path = os.path.join(source_path, f"{prefix}{epoch_index}", "stations.csv")
             data = pd.read_csv(input_path)
             data = data[["bikes", "trip_requirement", "fulfillment", "capacity"]].groupby(data["name"]).sum()
             data["fulfillment_ratio"] = list(
-                map(
-                    lambda x, y: round(x / (y + 1 / 1000), 4),
-                    data["fulfillment"],
-                    data["trip_requirement"]
-                )
+                map(lambda x, y: round(x / (y + 1 / 1000), 4), data["fulfillment"], data["trip_requirement"])
```
Comment (dongguangbin) [inline maro/cli/inspector/env_data_process.py:186]: mark

File: maro/communication/proxy.py [hunk 20]
```diff
@@ -171,8 +172,7 @@ def __init__(
 
             if isinstance(minimal_peers, int):
                 self._minimal_peers = {
-                    peer_type: max(minimal_peers, 1)
-                    for peer_type, peer_info in self._peers_info_dict.items()
+                    peer_type: max(minimal_peers, 1) for peer_type, peer_info in self._peers_info_dict.items()
```
Comment (Jinyu-W) [inline maro/communication/proxy.py:175]: start a new line for iteration or not ?

File: maro/cli/k8s/executors/k8s_executor.py [hunk 21]
```diff
@@ -411,9 +371,7 @@ def _standardize_schedule_details(start_schedule_deployment: dict) -> dict:
         with open(f"{K8sPaths.ABS_MARO_K8S_LIB}/deployments/internal/k8s_aks_start_schedule.yml") as fr:
             start_job_template = yaml.safe_load(fr)
         DeploymentValidator.validate_and_fill_dict(
-            template_dict=start_job_template,
-            actual_dict=start_schedule_deployment,
-            optional_key_to_value={}
+            template_dict=start_job_template, actual_dict=start_schedule_deployment, optional_key_to_value={}
```
Comment (dongguangbin) [inline maro/cli/k8s/executors/k8s_executor.py:374]: mark

File: maro/data_lib/cim/entities.py [hunk 22]
```diff
@@ -101,8 +103,13 @@ def __init__(self, tick: int, src_port_idx: int, dest_port_idx: int, quantity: i
         self.dest_port_idx = dest_port_idx
 
     def __repr__(self):
-        return "%s {tick: %r, src_port_idx: %r, dest_port_idx: %r, quantity: %r}" % \
-            (self.__class__.__name__, self.tick, self.src_port_idx, self.dest_port_idx, self.quantity)
+        return "%s {tick: %r, src_port_idx: %r, dest_port_idx: %r, quantity: %r}" % (
```
Comment (Jinyu-W) [inline maro/data_lib/cim/entities.py:106]: use f-string instead?

File: maro/cli/k8s/utils/k8s_manifest_generator.py [hunk 23]
```diff
@@ -15,79 +15,47 @@ def get_job_manifest(agent_pool_name: str, component_name: str, container_spec:
...
 def get_azurefile_volume_spec(name: str, share_name: str, secret_name: str):
-    return {
-        "name": name,
-        "azureFile": {
-            "secretName": secret_name,
-            "shareName": share_name,
-            "readOnly": False
-        }
-    }
+    return {"name": name, "azureFile": {"secretName": secret_name, "shareName": share_name, "readOnly": False}}
```
Comment (dongguangbin) [inline maro/cli/k8s/utils/k8s_manifest_generator.py:26]: mark

File: maro/rl/rollout/env_sampler.py [hunk 24]
```diff
@@ -171,9 +169,9 @@ def split_contents_by_agent(self) -> Dict[Any, ExpElement]:
                 reward_dict={agent_name: self.reward_dict[agent_name]},
                 terminal_dict={agent_name: self.terminal_dict[agent_name]},
                 next_state=self.next_state,
-                next_agent_state_dict={
-                    agent_name: self.next_agent_state_dict[agent_name]
-                } if self.next_agent_state_dict is not None and agent_name in self.next_agent_state_dict else {},
+                next_agent_state_dict={agent_name: self.next_agent_state_dict[agent_name]}
```
Comment (Jinyu-W) [inline maro/rl/rollout/env_sampler.py:172]: mark

File: maro/simulator/scenarios/cim/common.py [hunk 25]
```diff
@@ -9,13 +9,15 @@
...
+
     PARKING = 0
     SAILING = 1
 
 
 class ActionType(Enum):
     """Type of CIM action."""
-    LOAD = "load",
+
+    LOAD = ("load",)
```
Comment (Jinyu-W) [inline maro/simulator/scenarios/cim/common.py:20]: mark

File: maro/utils/exception/error_code.py [hunk 26]
```diff
@@ -19,11 +19,9 @@
     1011: "Peers Disconnection Error",
     1012: "MARO Send Again Error",
     1013: "Peers Rejoin Timeout",
-
```
Comment (Jinyu-W) [inline maro/utils/exception/error_code.py:22]: mark

File: maro/utils/logger.py [hunk 27]
```diff
@@ -111,14 +115,22 @@ class Logger(object):
...
+        stdout_level="INFO",
     ):
         self._file_format = FORMAT_NAME_TO_FILE_FORMAT[format_]
-        self._stdout_format = FORMAT_NAME_TO_STDOUT_FORMAT[format_] \
-            if format_ in FORMAT_NAME_TO_STDOUT_FORMAT else \
-            FORMAT_NAME_TO_FILE_FORMAT[format_]
-        self._stdout_level = os.environ.get('LOG_LEVEL') or stdout_level
+        self._stdout_format = (
+            FORMAT_NAME_TO_STDOUT_FORMAT[format_]
+            if format_ in FORMAT_NAME_TO_STDOUT_FORMAT
```
Comment (Jinyu-W) [inline maro/utils/logger.py:130]: strange that, it automatically add `()` here

File: maro/vector_env/env_process.py [hunk 28]
```diff
@@ -26,7 +26,7 @@ def __init__(self, pipe: Connection, *args, **kwargs):
     def run(self):
         """Initialize environment and process commands."""
         metrics = None
-        decision_event = None,
+        decision_event = (None,)
```
Comment (Jinyu-W) [inline maro/vector_env/env_process.py:29]: (duplicated) mark

File: maro/vector_env/vector_env.py [hunk 29]
```diff
@@ -155,7 +156,7 @@ def stop(self):
             for proc in self._sub_process_list:
                 proc.join()
 
-            for pipe in (self._env_pipes + self._pipes):
+            for pipe in self._env_pipes + self._pipes:
```
Comment (Jinyu-W) [inline maro/vector_env/vector_env.py:159]: mark

File: notebooks/vm_scheduling/vm_data_analytics.ipynb [hunk 30]
```diff
@@ -183,25 +183,32 @@
...
+    "vm_table = vm_table.replace({\"vmcorecountbucket\": \">24\"}, 30)\n",
+    "vm_table = vm_table.replace({\"vmmemorybucket\": \">64\"}, 70)\n",
     "\n",
-    "vm_table['vmcreated'] = pd.to_numeric(vm_table['vmcreated'], errors=\"coerce\", downcast=\"integer\") // 300\n",
-    "vm_table['vmdeleted'] = pd.to_numeric(vm_table['vmdeleted'], errors=\"coerce\", downcast=\"integer\") // 300\n",
-    "vm_table['vmcorecountbucket'] = pd.to_numeric(\n",
-    "    vm_table['vmcorecountbucket'], errors=\"coerce\", downcast=\"integer\"\n",
-    ")\n",
-    "vm_table['vmmemorybucket'] = pd.to_numeric(vm_table['vmmemorybucket'], errors=\"coerce\", downcast=\"integer\")\n",
+    "vm_table[\"vmcreated\"] = pd.to_numeric(vm_table[\"vmcreated\"], errors=\"coerce\", downcast=\"integer\") // 300\n",
```
Comment (Jinyu-W) [inline notebooks/vm_scheduling/vm_data_analytics.ipynb:204]: mark

File: scripts/random_config.py [hunk 31]
```diff
@@ -139,11 +139,20 @@
 """
 
 
-def generate_config(sku_num: int, supplier_num: int, warehouse_num: int, retailer_num: int, grid_width: int,
```
Comment (Jinyu-W) [inline scripts/random_config.py:142]: this file should be checked out from this branch

File: tests/cim/test_cim_scenario.py [hunk 32]
```diff
@@ -150,9 +149,52 @@ def test_vessel_movement(self) -> None:
...
 
             hard_coded_period = [
-                67, 75, 84, 67, 53, 58, 51, 58, 61, 49, 164, 182, 146, 164, 182, 146, 90, 98, 79, 95, 104, 84, 87, 97,
-                78, 154, 169, 136, 154, 169, 94, 105, 117, 94, 189, 210, 167, 189, 210, 167, 141, 158, 125, 141, 158,
-                125
+                67,
+                75,
+                84,
+                67,
+                53,
```
Comment (Jinyu-W) [inline tests/cim/test_cim_scenario.py:156]: mark ><

File: tests/cli/grass/test_grass_on_premises.py [hunk 33]
```diff
@@ -114,58 +113,52 @@ def _create_virtual_machines(cls, test_config: dict):
...
-        create_deployment["master"]["public_ip_address"] = (
-            ip_addresses[0]["virtualMachine"]["network"]["publicIpAddresses"][0]["ipAddress"]
-        )
-        create_deployment["master"]["private_ip_address"] = (
-            ip_addresses[0]["virtualMachine"]["network"]["privateIpAddresses"][0]
-        )
-        join_cluster_deployment["master"]["private_ip_address"] = (
-            ip_addresses[0]["virtualMachine"]["network"]["privateIpAddresses"][0]
-        )
+        create_deployment["master"]["public_ip_address"] = ip_addresses[0]["virtualMachine"]["network"][
```
Comment (Jinyu-W) [inline tests/cli/grass/test_grass_on_premises.py:137]:
```
... = (
    ...
)
```
?

File: maro/cli/utils/node_admin.py [hunk 34]
```diff
@@ -7,28 +7,22 @@
 
 
 def start_admin(*args, **kwargs):
-    print("""If got python module or file not found error, please run
+    print(
+        """If got python module or file not found error, please run
```
Comment (Jinyu-W) [inline maro/cli/utils/node_admin.py:11]: mark

**Comment by Jinyu-W** [comment]
Still some isort & flake8 issues.

Seems like it is a conflict?
```
2022-05-26 03:53:27 [ERROR]   Found errors in [flake8] linter!
2022-05-26 03:53:27 [ERROR]   Error code: 1. Command output:
------
/github/workspace/maro/simulator/scenarios/matrix_accessor.py:80:32: E203 whitespace before ':'
```

File: examples/vm_scheduling/offline_lp/vm_scheduling_ilp.py [hunk 35]
```diff
@@ -95,16 +96,20 @@ def _add_constraints(self, problem: LpProblem):
                         vm.core * self._mapping[pm_idx][vm_idx]
                         for vm_idx, vm in enumerate(self._future_vm_req)
                         if (vm.arrival_env_tick - self._env_tick <= t and vm.remaining_lifetime(self._env_tick) >= t)
-                    ) + self._pm_allocated_core[t][pm_idx] <= self._pm_capacity[pm_idx].core * self.core_upper_ratio,
-                    f"T{t}_PM{pm_idx}_core_capacity_limit"
+                    )
+                    + self._pm_allocated_core[t][pm_idx]
+                    <= self._pm_capacity[pm_idx].core * self.core_upper_ratio,
```
Comment (v-heli) [inline examples/vm_scheduling/offline_lp/vm_scheduling_ilp.py:101]: mark

File: maro/cli/inspector/citi_bike_dashboard.py [hunk 36]
```diff
@@ -306,33 +267,30 @@ def _generate_intra_view_by_station(
...
-        color="Attributes",
-        tooltip=["Attributes", "Count", "snapshot_index"]
-    ).properties(
-        width=700,
-        height=380
+    data_filtered.rename(columns={"frame_index": "snapshot_index"}, inplace=True)
+    data_melt = data_filtered.melt("snapshot_index", var_name="Attributes", value_name="Count")
+    intra_station_line_chart = (
+        alt.Chart(data_melt)
+        .mark_line()
```
Comment (v-heli) [inline maro/cli/inspector/citi_bike_dashboard.py:274]: mark

File: maro/cli/k8s/executors/k8s_aks_executor.py [hunk 37]
```diff
@@ -225,10 +217,10 @@ def _create_storage_account_secret(cluster_details: dict) -> None:
                 metadata=client.V1ObjectMeta(name="azure-storage-account-secret"),
                 data={
                     "azurestorageaccountname": base64.b64encode(storage_account_name.encode()).decode(),
-                    "azurestorageaccountkey": base64.b64encode(bytes(storage_key.encode())).decode()
-                }
+                    "azurestorageaccountkey": base64.b64encode(bytes(storage_key.encode())).decode(),
+                },
```
Comment (v-heli) [inline maro/cli/k8s/executors/k8s_aks_executor.py:221]: mark

File: maro/cli/maro.py [hunk 38]
```diff
@@ -289,42 +261,35 @@ def load_parser_grass(prev_parser: ArgumentParser, global_parser: ArgumentParser
...
 
     # maro grass node scale
     from maro.cli.grass.node import scale_node
+
     parser_node_scale = parser_node_subparsers.add_parser(
         "scale",
         help="Scale up or scale down nodes to target number",
         examples=CliExamples.MARO_GRASS_NODE_SCALE,
-        parents=[global_parser]
+        parents=[global_parser],
```
Comment (v-heli) [inline maro/cli/maro.py:292]: mark

File: maro/cli/utils/subprocess.py [hunk 39]
```diff
@@ -24,15 +24,15 @@ def run(command: str, timeout: int = None) -> str:
             str: return stdout of the command.
         """
         if platform.system() == "Windows":
-            command = f"powershell.exe -Command \"{command}\""
+            command = f'powershell.exe -Command "{command}"'
```
Comment (v-heli) [inline maro/cli/utils/subprocess.py:27]: mark

File: maro/data_lib/binary_converter.py [hunk 40]
```diff
@@ -38,7 +38,7 @@ def convert_val(val: str, dtype: str, tzone):
 
     # process the value first
     # clear
-    val = val.strip("\"\'")
+    val = val.strip("\"'")
```
Comment (v-heli) [inline maro/data_lib/binary_converter.py:41]: mark

File: examples/cim/rl/config.py (hunk 6 above)
Comment (lihuoran) [reply examples/cim/rl/config.py:4]: Ignore this file in the command.

File: examples/cim/rl/config.py (hunk 7 above)
Comment (lihuoran) [reply examples/cim/rl/config.py:25]: Add related descriptions into wiki.

File: maro/cli/inspector/cim_dashboard.py [hunk 41]
```diff
@@ -610,24 +547,12 @@ def _generate_top_k_summary(data: pd.DataFrame, snapshot_index: int, index_name_
     """
     data_summary = data[data["frame_index"] == snapshot_index].reset_index(drop=True)
     data_summary["fulfillment_ratio"] = list(
-        map(
-            lambda x, y: round(x / (y + 1 / 1000), 4),
-            data_summary["acc_fulfillment"],
-            data_summary["acc_booking"]
-        )
+        map(lambda x, y: round(x / (y + 1 / 1000), 4), data_summary["acc_fulfillment"], data_summary["acc_booking"])
```
[the remark this answers is not in the document]
Comment (lihuoran) [reply maro/cli/inspector/cim_dashboard.py:550]: [TBD] Can we ignore specific functions (map, dict, etc.)?

File: maro/cli/grass/lib/services/node_agent/agent.py (hunk 15 above)
Comment (lihuoran) [reply maro/cli/grass/lib/services/node_agent/agent.py:371]: Shorten variable name.

File: maro/rl/rollout/env_sampler.py (hunk 24 above)
Comment (lihuoran) [reply maro/rl/rollout/env_sampler.py:172]: [TBD] Do not DINGGE.

File: maro/simulator/scenarios/cim/common.py (hunk 25 above)
Comment (lihuoran) [reply maro/simulator/scenarios/cim/common.py:20]: Fix this.

File: maro/utils/exception/error_code.py (hunk 26 above)
Comment (lihuoran) [reply maro/utils/exception/error_code.py:22]: Exclude this file.

File: tests/cim/test_cim_scenario.py (hunk 32 above)
Comment (lihuoran) [reply tests/cim/test_cim_scenario.py:156]: Add exclude mark.

File: tests/cli/grass/test_grass_on_premises.py (hunk 33 above)
Comment (lihuoran) [reply tests/cli/grass/test_grass_on_premises.py:137]: [TBD] Do not start newline after `[`.

File: maro/cli/utils/node_admin.py (hunk 34 above)
Comment (lihuoran) [reply maro/cli/utils/node_admin.py:11]: Fix the source code.

File: examples/vector_env/hello.py (hunk 10 above)
Comment (lihuoran) [reply examples/vector_env/hello.py:33]: See [this doc](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#slices) for explanations. I personally buy it.

File: maro/cli/grass/lib/services/node_agent/agent.py (hunk 16 above)
Comment (lihuoran) [reply maro/cli/grass/lib/services/node_agent/agent.py:393]: See [this](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#trailing-commas)

**Comment by lihuoran** [comment]
> /github/workspace/maro/simulator/scenarios/matrix_accessor.py:80:32: E203 whitespace before ':'

See this [discussion](https://github.com/psf/black/issues/354#issuecomment-397685631)

**Pull request closed without merging**